AMM/project/scripts/atomic-add-liquidity/tests/cases/revert.rs

```diff
@@ -0,0 +1,134 @@
+use crate::utils::{expected_liquidity, setup};
+use fuels::prelude::*;
+use test_utils::interface::{
+    atomic_add_liquidity_script_mod::{Asset, AssetPair},
+    LiquidityParameters, SCRIPT_GAS_LIMIT,
+};
+
+#[tokio::test]
+#[should_panic(expected = "DesiredLiquidityZero")]
+async fn when_desired_liquidity_zero() {
+    let (script_instance, exchange, liquidity_parameters, _transaction_parameters) =
+        setup((1000, 1000), 1000).await;
+
+    script_instance
+        .main(
+            exchange.id,
+            LiquidityParameters {
+                deposits: AssetPair {
+                    a: Asset {
+                        id: ContractId::new(*exchange.pair.0),
+                        amount: liquidity_parameters.amounts.0,
+                    },
+                    b: Asset {
+                        id: ContractId::new(*exchange.pair.1),
+                        amount: liquidity_parameters.amounts.1,
+                    },
+                },
+                liquidity: 0, // desired liquidity is 0
+                deadline: liquidity_parameters.deadline,
+            },
+        )
+        .call()
+        .await
+        .unwrap();
+}
+
+#[tokio::test]
+#[should_panic(expected = "Revert(18446744073709486080)")]
+// the contract call in the script fails with "DesiredAmountTooHigh" but that message is not propagated
```

Comment:
Noting that we need to wait for a new PR / new release for this to be propagated. Could use a branch until then via `staging`